jobs: TestRegistryLifecycle failed

[(jobs).TestRegistryLifecycle failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1750798&tab=buildLog) on [master@5c37cf2d12bafd4a01d4ac7f1725a0ae6e498c9c](https://github.com/cockroachdb/cockroach/commits/5c37cf2d12bafd4a01d4ac7f1725a0ae6e498c9c):

```
=== RUN   TestRegistryLifecycle/cancel
I200215 02:29:49.146140 5234 base/addr_validation.go:296  [n?] server certificate addresses: IP=127.0.0.1,::1; DNS=localhost,*.local; CN=node
I200215 02:29:49.146168 5234 base/addr_validation.go:342  [n?] web UI certificate addresses: IP=127.0.0.1,::1; DNS=localhost,*.local; CN=node
W200215 02:29:49.168950 5234 server/status/runtime.go:308  [n?] Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I200215 02:29:49.176609 5234 server/server.go:998  [n?] monitoring forward clock jumps based on server.clock.forward_jump_check_enabled
I200215 02:29:49.192917 5234 server/config.go:566  [n?] 1 storage engine initialized
I200215 02:29:49.192944 5234 server/config.go:569  [n?] RocksDB cache size: 128 MiB
I200215 02:29:49.192954 5234 server/config.go:569  [n?] store 0: in-memory, size 0 B
I200215 02:29:49.193064 5234 server/server.go:1453  [n?] no stores bootstrapped and --join flag specified, awaiting init command or join with an already initialized node.
I200215 02:29:49.193107 5234 server/server.go:1463  [n?] **** add additional nodes by specifying --join=127.0.0.1:43003
I200215 02:29:49.205987 5234 server/node.go:334  [n?] **** cluster ca4080cc-be93-4c8e-b5fb-0c9b92342674 has been created
I200215 02:29:49.206289 5234 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:43003" > attrs:<> locality:<> ServerVersion:<major_val:19 minor_val:2 patch:0 unstable:12 > build_tag:"v20.1.0-alpha.20191118-2194-g5c37cf2" started_at:1581733789206198749 cluster_name:"" sql_address:<network_field:"tcp" address_field:"127.0.0.1:41857" > 
I200215 02:29:49.224124 5234 server/node.go:438  [n1] initialized store [n1,s1]: disk (capacity=512 MiB, available=512 MiB, used=0 B, logicalBytes=13 KiB), ranges=28, leases=0, queries=0.00, writes=0.00, bytesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=950.00 pMax=11278.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00 pMax=0.00}
I200215 02:29:49.224374 5234 storage/stores.go:234  [n1] read 0 node addresses from persistent storage
W200215 02:29:49.224467 5364 storage/store.go:1538  [n1,s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I200215 02:29:49.224534 5234 server/node.go:655  [n1] connecting to gossip network to verify cluster ID...
I200215 02:29:49.224826 5234 server/node.go:675  [n1] node connected via gossip and verified as part of cluster "ca4080cc-be93-4c8e-b5fb-0c9b92342674"
I200215 02:29:49.225280 5234 server/node.go:522  [n1] node=1: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I200215 02:29:49.225668 5234 server/server.go:2141  [n1] Could not start goroutine dumper worker due to: directory to store dumps could not be determined
I200215 02:29:49.225907 5234 server/server.go:1583  [n1] starting https server at 127.0.0.1:44243 (use: 127.0.0.1:44243)
I200215 02:29:49.226079 5234 server/server.go:1588  [n1] starting postgres server at 127.0.0.1:41857 (use: 127.0.0.1:41857)
I200215 02:29:49.226110 5234 server/server.go:1590  [n1] starting grpc server at 127.0.0.1:43003
I200215 02:29:49.226433 5234 server/server.go:1591  [n1] advertising CockroachDB node at 127.0.0.1:43003
I200215 02:29:49.312044 5475 sql/event_log.go:132  [n1,intExec=optInToDiagnosticsStatReporting] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:root}
I200215 02:29:49.334785 4219 sql/event_log.go:132  [n1,intExec=set-setting] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:19.2-12 User:root}
I200215 02:29:49.351946 5822 sql/event_log.go:132  [n1,intExec=initializeClusterSecret] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.secret Value:2561729b-fc5c-44dd-8a06-4886ab95d323 User:root}
I200215 02:29:49.358396 5829 sql/event_log.go:132  [n1,intExec=create-default-db] Event: "create_database", target: 50, info: {DatabaseName:defaultdb Statement:CREATE DATABASE IF NOT EXISTS defaultdb User:root}
I200215 02:29:49.364123 5906 sql/event_log.go:132  [n1,intExec=create-default-db] Event: "create_database", target: 51, info: {DatabaseName:postgres Statement:CREATE DATABASE IF NOT EXISTS postgres User:root}
I200215 02:29:49.427231 5234 server/server.go:1665  [n1] done ensuring all necessary migrations have run
I200215 02:29:49.427279 5234 server/server.go:1946  [n1] serving sql connections
I200215 02:29:49.432756 6065 sql/event_log.go:132  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:127.0.0.1:43003 Attrs: Locality: ServerVersion:19.2-12 BuildTag:v20.1.0-alpha.20191118-2194-g5c37cf2 StartedAt:1581733789206198749 LocalityAddress:[] ClusterName: SQLAddress:127.0.0.1:41857} ClusterID:ca4080cc-be93-4c8e-b5fb-0c9b92342674 StartedAt:1581733789206198749 LastUp:1581733789206198749}
I200215 02:29:49.434951 6063 server/server_update.go:55  [n1] no need to upgrade, cluster already at the newest version
I200215 02:29:49.444053 5839 jobs/registry.go:669  [n1] job 529738594492678145: stepping through state running with error <nil>
I200215 02:29:49.483725 5234 util/stop/stopper.go:539  quiescing
I200215 02:29:49.483868 5839 jobs/registry.go:669  [n1] job 529738594492678145: stepping through state succeeded with error <nil>
W200215 02:29:49.483922 5839 internal/client/txn.go:566  [n1] failure aborting transaction: node unavailable; try another peer; abort caused by: context canceled
E200215 02:29:49.483958 6101 jobs/registry.go:407  error while adopting jobs: job 529738594492678145: can't query system.jobs: job-status: failed to send RPC: sending to all 1 replicas failed; last error: (err: node unavailable; try another peer) <nil>
I200215 02:29:49.483959 5839 jobs/registry.go:669  [n1] job 529738594492678145: stepping through state reverting with error could not mark job 529738594492678145 as succeeded: context canceled
W200215 02:29:49.484013 5839 internal/client/txn.go:566  [n1] failure aborting transaction: node unavailable; try another peer; abort caused by: context canceled
E200215 02:29:49.484113 5839 jobs/registry.go:802  [n1] job 529738594492678145: adoption completed with error job 529738594492678145: could not mark as reverting: could not mark job 529738594492678145 as succeeded: context canceled: context canceled
W200215 02:29:49.484132 5839 internal/client/txn.go:566  [n1] failure aborting transaction: node unavailable; try another peer; abort caused by: context canceled
E200215 02:29:49.484142 5839 jobs/registry.go:806  [n1] job 529738594492678145: failed querying status: context canceled
W200215 02:29:49.484883 6082 storage/intentresolver/intent_resolver.go:838  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
E200215 02:29:49.485737 6101 jobs/registry.go:407  error while adopting jobs: failed querying for jobs: adopt-job: node unavailable; try another peer
E200215 02:29:49.489311 6101 jobs/registry.go:407  error while adopting jobs: failed querying for jobs: adopt-job: node unavailable; try another peer
E200215 02:29:49.490997 6101 jobs/registry.go:407  error while adopting jobs: failed querying for jobs: adopt-job: node unavailable; try another peer
    --- FAIL: TestRegistryLifecycle/cancel (0.35s)
        jobs_test.go:189: Starting resume
        jobs_test.go:378: unexpected <nil>
        jobs_test.go:197: Exiting resume

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRegistryLifecycle PKG=./pkg/jobs TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRegistryLifecycle.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(jobs).TestRegistryLifecycle failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1751902&tab=buildLog) on [master@bb14400cdd1397f47c3781b4830199f169cef497](https://github.com/cockroachdb/cockroach/commits/bb14400cdd1397f47c3781b4830199f169cef497):

```
=== RUN   TestRegistryLifecycle
--- FAIL: TestRegistryLifecycle (4.33s)
=== RUN   TestRegistryLifecycle/cancel
I200216 06:17:01.503371 5186 base/addr_validation.go:296  [n?] server certificate addresses: IP=127.0.0.1,::1; DNS=localhost,*.local; CN=node
I200216 06:17:01.503399 5186 base/addr_validation.go:342  [n?] web UI certificate addresses: IP=127.0.0.1,::1; DNS=localhost,*.local; CN=node
W200216 06:17:01.546331 5186 server/status/runtime.go:308  [n?] Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I200216 06:17:01.555968 5186 server/server.go:998  [n?] monitoring forward clock jumps based on server.clock.forward_jump_check_enabled
I200216 06:17:01.590310 5186 server/config.go:566  [n?] 1 storage engine initialized
I200216 06:17:01.590338 5186 server/config.go:569  [n?] RocksDB cache size: 128 MiB
I200216 06:17:01.590348 5186 server/config.go:569  [n?] store 0: in-memory, size 0 B
I200216 06:17:01.590477 5186 server/server.go:1453  [n?] no stores bootstrapped and --join flag specified, awaiting init command or join with an already initialized node.
I200216 06:17:01.590739 5186 server/server.go:1463  [n?] **** add additional nodes by specifying --join=127.0.0.1:32929
I200216 06:17:01.599325 5186 server/node.go:334  [n?] **** cluster f41f89a1-1195-4c53-bc88-8e464be815af has been created
I200216 06:17:01.599900 5186 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:32929" > attrs:<> locality:<> ServerVersion:<major_val:19 minor_val:2 patch:0 unstable:12 > build_tag:"v20.1.0-alpha.20191118-2202-gbb14400" started_at:1581833821599810609 cluster_name:"" sql_address:<network_field:"tcp" address_field:"127.0.0.1:45729" > 
W200216 06:17:01.605193 5295 storage/store.go:1538  [n1,s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I200216 06:17:01.612116 5186 server/node.go:438  [n1] initialized store [n1,s1]: disk (capacity=512 MiB, available=512 MiB, used=0 B, logicalBytes=13 KiB), ranges=28, leases=2, queries=0.00, writes=0.00, bytesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=950.00 pMax=11278.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00 pMax=0.00}
I200216 06:17:01.612463 5186 storage/stores.go:234  [n1] read 0 node addresses from persistent storage
I200216 06:17:01.613131 5186 server/node.go:655  [n1] connecting to gossip network to verify cluster ID...
I200216 06:17:01.613417 5186 server/node.go:675  [n1] node connected via gossip and verified as part of cluster "f41f89a1-1195-4c53-bc88-8e464be815af"
I200216 06:17:01.613816 5186 server/node.go:522  [n1] node=1: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I200216 06:17:01.614466 5186 server/server.go:2141  [n1] Could not start goroutine dumper worker due to: directory to store dumps could not be determined
I200216 06:17:01.616152 5186 server/server.go:1583  [n1] starting https server at 127.0.0.1:39485 (use: 127.0.0.1:39485)
I200216 06:17:01.616213 5186 server/server.go:1588  [n1] starting postgres server at 127.0.0.1:45729 (use: 127.0.0.1:45729)
I200216 06:17:01.616239 5186 server/server.go:1590  [n1] starting grpc server at 127.0.0.1:32929
I200216 06:17:01.616259 5186 server/server.go:1591  [n1] advertising CockroachDB node at 127.0.0.1:32929
I200216 06:17:01.690958 5343 sql/event_log.go:132  [n1,intExec=optInToDiagnosticsStatReporting] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:root}
I200216 06:17:01.701295 5507 sql/event_log.go:132  [n1,intExec=set-setting] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:19.2-12 User:root}
I200216 06:17:01.725745 5600 sql/event_log.go:132  [n1,intExec=initializeClusterSecret] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.secret Value:c83103d0-7551-4283-9260-fc09fba5f1a4 User:root}
I200216 06:17:01.755578 5735 sql/event_log.go:132  [n1,intExec=create-default-db] Event: "create_database", target: 50, info: {DatabaseName:defaultdb Statement:CREATE DATABASE IF NOT EXISTS defaultdb User:root}
I200216 06:17:01.762306 5720 sql/event_log.go:132  [n1,intExec=create-default-db] Event: "create_database", target: 51, info: {DatabaseName:postgres Statement:CREATE DATABASE IF NOT EXISTS postgres User:root}
I200216 06:17:01.781337 5186 server/server.go:1665  [n1] done ensuring all necessary migrations have run
I200216 06:17:01.781383 5186 server/server.go:1946  [n1] serving sql connections
I200216 06:17:01.790098 5993 sql/event_log.go:132  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:127.0.0.1:32929 Attrs: Locality: ServerVersion:19.2-12 BuildTag:v20.1.0-alpha.20191118-2202-gbb14400 StartedAt:1581833821599810609 LocalityAddress:[] ClusterName: SQLAddress:127.0.0.1:45729} ClusterID:f41f89a1-1195-4c53-bc88-8e464be815af StartedAt:1581833821599810609 LastUp:1581833821599810609}
I200216 06:17:01.797531 5991 server/server_update.go:55  [n1] no need to upgrade, cluster already at the newest version
I200216 06:17:01.809931 5451 jobs/registry.go:669  [n1] job 530066380495585281: stepping through state running with error <nil>
I200216 06:17:01.885482 5186 util/stop/stopper.go:539  quiescing
W200216 06:17:01.885869 5444 storage/intentresolver/intent_resolver.go:838  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
E200216 06:17:01.890167 5451 jobs/registry.go:802  [n1] job 530066380495585281: adoption completed with error job 530066380495585281: node liveness error: restarting in background
W200216 06:17:01.890266 5451 internal/client/txn.go:566  [n1] failure aborting transaction: node unavailable; try another peer; abort caused by: context canceled
E200216 06:17:01.890281 5451 jobs/registry.go:806  [n1] job 530066380495585281: failed querying status: context canceled
E200216 06:17:01.892013 5997 jobs/registry.go:407  error while adopting jobs: job 530066380495585281: can't query system.jobs: job-status: failed to send RPC: sending to all 1 replicas failed; last error: (err: node unavailable; try another peer) <nil>
    --- FAIL: TestRegistryLifecycle/cancel (0.40s)
        jobs_test.go:189: Starting resume
        jobs_test.go:378: unexpected <nil>
        jobs_test.go:197: Exiting resume

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRegistryLifecycle PKG=./pkg/jobs TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRegistryLifecycle.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(jobs).TestRegistryLifecycle failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1754448&tab=buildLog) on [master@cfb34abbfff493b16da81057e5b422ae6eeaf558](https://github.com/cockroachdb/cockroach/commits/cfb34abbfff493b16da81057e5b422ae6eeaf558):

```
=== RUN   TestRegistryLifecycle
--- FAIL: TestRegistryLifecycle (4.14s)
=== RUN   TestRegistryLifecycle/cancel
I200218 17:26:58.342631 4894 base/addr_validation.go:296  [n?] server certificate addresses: IP=127.0.0.1,::1; DNS=localhost,*.local; CN=node
I200218 17:26:58.342673 4894 base/addr_validation.go:342  [n?] web UI certificate addresses: IP=127.0.0.1,::1; DNS=localhost,*.local; CN=node
W200218 17:26:58.364236 4894 server/status/runtime.go:308  [n?] Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I200218 17:26:58.387033 4894 server/server.go:998  [n?] monitoring forward clock jumps based on server.clock.forward_jump_check_enabled
I200218 17:26:58.425819 4894 server/config.go:566  [n?] 1 storage engine initialized
I200218 17:26:58.425843 4894 server/config.go:569  [n?] RocksDB cache size: 128 MiB
I200218 17:26:58.425853 4894 server/config.go:569  [n?] store 0: in-memory, size 0 B
I200218 17:26:58.425974 4894 server/server.go:1453  [n?] no stores bootstrapped and --join flag specified, awaiting init command or join with an already initialized node.
I200218 17:26:58.426011 4894 server/server.go:1463  [n?] **** add additional nodes by specifying --join=127.0.0.1:39655
I200218 17:26:58.430502 4894 server/node.go:334  [n?] **** cluster f60f2ff4-b15c-4fa9-aeab-652141f9ddfb has been created
I200218 17:26:58.430679 4894 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:39655" > attrs:<> locality:<> ServerVersion:<major_val:19 minor_val:2 patch:0 unstable:12 > build_tag:"v20.1.0-alpha.20191118-2218-gcfb34ab" started_at:1582046818430602605 cluster_name:"" sql_address:<network_field:"tcp" address_field:"127.0.0.1:41137" > 
W200218 17:26:58.435054 5092 storage/store.go:1538  [n1,s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I200218 17:26:58.435414 4894 server/node.go:438  [n1] initialized store [n1,s1]: disk (capacity=512 MiB, available=512 MiB, used=0 B, logicalBytes=13 KiB), ranges=28, leases=0, queries=0.00, writes=0.00, bytesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=950.00 pMax=11278.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00 pMax=0.00}
I200218 17:26:58.435503 4894 storage/stores.go:234  [n1] read 0 node addresses from persistent storage
I200218 17:26:58.448479 4894 server/node.go:655  [n1] connecting to gossip network to verify cluster ID...
I200218 17:26:58.448527 4894 server/node.go:675  [n1] node connected via gossip and verified as part of cluster "f60f2ff4-b15c-4fa9-aeab-652141f9ddfb"
I200218 17:26:58.448680 4894 server/node.go:522  [n1] node=1: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I200218 17:26:58.448767 4894 server/server.go:2141  [n1] Could not start goroutine dumper worker due to: directory to store dumps could not be determined
I200218 17:26:58.448829 4894 server/server.go:1583  [n1] starting https server at 127.0.0.1:36585 (use: 127.0.0.1:36585)
I200218 17:26:58.448859 4894 server/server.go:1588  [n1] starting postgres server at 127.0.0.1:41137 (use: 127.0.0.1:41137)
I200218 17:26:58.448877 4894 server/server.go:1590  [n1] starting grpc server at 127.0.0.1:39655
I200218 17:26:58.448905 4894 server/server.go:1591  [n1] advertising CockroachDB node at 127.0.0.1:39655
I200218 17:26:58.541729 5221 sql/event_log.go:132  [n1,intExec=optInToDiagnosticsStatReporting] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:root}
I200218 17:26:58.566914 5233 sql/event_log.go:132  [n1,intExec=set-setting] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:19.2-12 User:root}
I200218 17:26:58.585791 5542 sql/event_log.go:132  [n1,intExec=initializeClusterSecret] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.secret Value:cd71b998-a197-4f6d-94a5-8346c861499e User:root}
I200218 17:26:58.594152 5606 sql/event_log.go:132  [n1,intExec=create-default-db] Event: "create_database", target: 50, info: {DatabaseName:defaultdb Statement:CREATE DATABASE IF NOT EXISTS defaultdb User:root}
I200218 17:26:58.619500 5637 sql/event_log.go:132  [n1,intExec=create-default-db] Event: "create_database", target: 51, info: {DatabaseName:postgres Statement:CREATE DATABASE IF NOT EXISTS postgres User:root}
I200218 17:26:58.694181 4894 server/server.go:1665  [n1] done ensuring all necessary migrations have run
I200218 17:26:58.694260 4894 server/server.go:1946  [n1] serving sql connections
I200218 17:26:58.697977 5830 server/server_update.go:55  [n1] no need to upgrade, cluster already at the newest version
I200218 17:26:58.701282 5832 sql/event_log.go:132  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:127.0.0.1:39655 Attrs: Locality: ServerVersion:19.2-12 BuildTag:v20.1.0-alpha.20191118-2218-gcfb34ab StartedAt:1582046818430602605 LocalityAddress:[] ClusterName: SQLAddress:127.0.0.1:41137} ClusterID:f60f2ff4-b15c-4fa9-aeab-652141f9ddfb StartedAt:1582046818430602605 LastUp:1582046818430602605}
I200218 17:26:58.795273 5850 jobs/registry.go:669  [n1] job 530764328830959617: stepping through state running with error <nil>
I200218 17:26:58.869423 5836 jobs/jobs.go:612  job 530764328830959617: adopting
I200218 17:26:58.873561 5836 jobs/registry.go:1017  job 530764328830959617: resuming execution
I200218 17:26:58.875090 4894 util/stop/stopper.go:539  quiescing
W200218 17:26:58.875906 5811 storage/intentresolver/intent_resolver.go:838  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
E200218 17:26:58.876461 5836 jobs/registry.go:407  error while adopting jobs: node unavailable; try another peer
E200218 17:26:58.877221 5850 jobs/registry.go:802  [n1] job 530764328830959617: adoption completed with error job 530764328830959617: node liveness error: restarting in background
W200218 17:26:58.877330 5850 internal/client/txn.go:566  [n1] failure aborting transaction: node unavailable; try another peer; abort caused by: context canceled
E200218 17:26:58.877586 5850 jobs/registry.go:806  [n1] job 530764328830959617: failed querying status: context canceled
    --- FAIL: TestRegistryLifecycle/cancel (0.56s)
        jobs_test.go:189: Starting resume
        jobs_test.go:378: unexpected <nil>
        jobs_test.go:197: Exiting resume

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRegistryLifecycle PKG=./pkg/jobs TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRegistryLifecycle.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(jobs).TestRegistryLifecycle failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1755071&tab=buildLog) on [master@5e226eec1864718c45f058a9f9e3ff1e7d649d31](https://github.com/cockroachdb/cockroach/commits/5e226eec1864718c45f058a9f9e3ff1e7d649d31):

```
=== RUN   TestRegistryLifecycle
--- FAIL: TestRegistryLifecycle (2.98s)
=== RUN   TestRegistryLifecycle/cancel
I200219 03:29:00.605956 5109 base/addr_validation.go:296  [n?] server certificate addresses: IP=127.0.0.1,::1; DNS=localhost,*.local; CN=node
I200219 03:29:00.605984 5109 base/addr_validation.go:342  [n?] web UI certificate addresses: IP=127.0.0.1,::1; DNS=localhost,*.local; CN=node
W200219 03:29:00.621663 5109 server/status/runtime.go:308  [n?] Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I200219 03:29:00.630517 5109 server/server.go:998  [n?] monitoring forward clock jumps based on server.clock.forward_jump_check_enabled
I200219 03:29:00.656787 5109 server/config.go:566  [n?] 1 storage engine initialized
I200219 03:29:00.656815 5109 server/config.go:569  [n?] RocksDB cache size: 128 MiB
I200219 03:29:00.656824 5109 server/config.go:569  [n?] store 0: in-memory, size 0 B
I200219 03:29:00.656953 5109 server/server.go:1453  [n?] no stores bootstrapped and --join flag specified, awaiting init command or join with an already initialized node.
I200219 03:29:00.656990 5109 server/server.go:1463  [n?] **** add additional nodes by specifying --join=127.0.0.1:43847
I200219 03:29:00.661415 5109 server/node.go:334  [n?] **** cluster 052da3f8-953e-4cf7-8fd9-981e7c9a0f7f has been created
I200219 03:29:00.661582 5109 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:43847" > attrs:<> locality:<> ServerVersion:<major_val:19 minor_val:2 patch:0 unstable:12 > build_tag:"v20.1.0-beta.1-177-g5e226ee" started_at:1582082940661508358 cluster_name:"" sql_address:<network_field:"tcp" address_field:"127.0.0.1:45255" > 
I200219 03:29:00.665214 5109 server/node.go:438  [n1] initialized store [n1,s1]: disk (capacity=512 MiB, available=512 MiB, used=0 B, logicalBytes=13 KiB), ranges=28, leases=0, queries=0.00, writes=0.00, bytesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=950.00 pMax=11278.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00 pMax=0.00}
I200219 03:29:00.665288 5109 storage/stores.go:234  [n1] read 0 node addresses from persistent storage
I200219 03:29:00.665352 5109 server/node.go:655  [n1] connecting to gossip network to verify cluster ID...
I200219 03:29:00.665382 5109 server/node.go:675  [n1] node connected via gossip and verified as part of cluster "052da3f8-953e-4cf7-8fd9-981e7c9a0f7f"
I200219 03:29:00.665428 5109 server/node.go:522  [n1] node=1: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I200219 03:29:00.665485 5109 server/server.go:2141  [n1] Could not start goroutine dumper worker due to: directory to store dumps could not be determined
I200219 03:29:00.665534 5109 server/server.go:1583  [n1] starting https server at 127.0.0.1:34923 (use: 127.0.0.1:34923)
I200219 03:29:00.665562 5109 server/server.go:1588  [n1] starting postgres server at 127.0.0.1:45255 (use: 127.0.0.1:45255)
I200219 03:29:00.665579 5109 server/server.go:1590  [n1] starting grpc server at 127.0.0.1:43847
I200219 03:29:00.665597 5109 server/server.go:1591  [n1] advertising CockroachDB node at 127.0.0.1:43847
W200219 03:29:00.666040 5239 storage/store.go:1538  [n1,s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I200219 03:29:00.694724 5100 sql/event_log.go:132  [n1,intExec=optInToDiagnosticsStatReporting] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:root}
I200219 03:29:00.712259 5308 sql/event_log.go:132  [n1,intExec=set-setting] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:19.2-12 User:root}
I200219 03:29:00.727669 5640 sql/event_log.go:132  [n1,intExec=initializeClusterSecret] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.secret Value:72e85dea-bee3-4724-9550-66e02fa2e9e8 User:root}
I200219 03:29:00.734558 5553 sql/event_log.go:132  [n1,intExec=create-default-db] Event: "create_database", target: 50, info: {DatabaseName:defaultdb Statement:CREATE DATABASE IF NOT EXISTS defaultdb User:root}
I200219 03:29:00.739757 5733 sql/event_log.go:132  [n1,intExec=create-default-db] Event: "create_database", target: 51, info: {DatabaseName:postgres Statement:CREATE DATABASE IF NOT EXISTS postgres User:root}
I200219 03:29:00.756309 5109 server/server.go:1665  [n1] done ensuring all necessary migrations have run
I200219 03:29:00.756858 5109 server/server.go:1946  [n1] serving sql connections
I200219 03:29:00.759608 5972 server/server_update.go:55  [n1] no need to upgrade, cluster already at the newest version
I200219 03:29:00.762331 5974 sql/event_log.go:132  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:127.0.0.1:43847 Attrs: Locality: ServerVersion:19.2-12 BuildTag:v20.1.0-beta.1-177-g5e226ee StartedAt:1582082940661508358 LocalityAddress:[] ClusterName: SQLAddress:127.0.0.1:45255} ClusterID:052da3f8-953e-4cf7-8fd9-981e7c9a0f7f StartedAt:1582082940661508358 LastUp:1582082940661508358}
I200219 03:29:00.793524 6010 jobs/registry.go:669  [n1] job 530882693580685313: stepping through state running with error <nil>
I200219 03:29:00.831064 5109 util/stop/stopper.go:539  quiescing
E200219 03:29:00.831662 6010 jobs/registry.go:802  [n1] job 530882693580685313: adoption completed with error job 530882693580685313: node liveness error: restarting in background
W200219 03:29:00.832479 6010 internal/client/txn.go:566  [n1] failure aborting transaction: node unavailable; try another peer; abort caused by: context canceled
E200219 03:29:00.832501 6010 jobs/registry.go:806  [n1] job 530882693580685313: failed querying status: context canceled
W200219 03:29:00.831710 5869 storage/intentresolver/intent_resolver.go:838  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
E200219 03:29:00.832345 5978 jobs/registry.go:407  error while adopting jobs: job 530882693580685313: can't query system.jobs: job-status: failed to send RPC: sending to all 1 replicas failed; last error: (err: node unavailable; try another peer) <nil>
    --- FAIL: TestRegistryLifecycle/cancel (0.23s)
        jobs_test.go:189: Starting resume
        jobs_test.go:378: unexpected <nil>
        jobs_test.go:197: Exiting resume

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRegistryLifecycle PKG=./pkg/jobs TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRegistryLifecycle.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/jobs/jobs_test.go
+++ b/pkg/jobs/jobs_test.go
@@ -373,17 +373,15 @@ func TestRegistryLifecycle(t *testing.T) {
 		rts.mu.e.ResumeStart = true
 		rts.check(t, jobs.StatusRunning)
 		rts.sqlDB.Exec(t, "CANCEL JOB $1", *job.ID())
-		// Test for a canceled error message.
-		if err := job.CheckStatus(rts.ctx); !testutils.IsError(err, "cannot update progress") {
-			t.Fatalf("unexpected %v", err)
-		}
 		rts.mu.e.OnFailOrCancelStart = true
 		rts.failOrCancelCheckCh <- struct{}{}
+		// Test for Reverting status.
 		rts.check(t, jobs.StatusReverting)
 		rts.mu.e.OnFailOrCancelExit++
 		rts.failOrCancelCh <- nil
 		rts.mu.e.Terminal++
 		rts.termCh <- struct{}{}
+		// Test for Canceled status.
 		rts.check(t, jobs.StatusCanceled)
 	})
 

